src/java/fr/paris/lutece/plugins/workflow/modules/formspdf/business/FormsPDFTaskTemplateDAO.java

```diff
@@ -0,0 +1,113 @@
+package fr.paris.lutece.plugins.workflow.modules.formspdf.business;
+
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+
+import fr.paris.lutece.plugins.workflow.modules.formspdf.service.FormsPDFPlugin;
+import fr.paris.lutece.util.sql.DAOUtil;
+
+public class FormsPDFTaskTemplateDAO implements IFormsPDFTaskTemplateDAO {
+	
+	 private static final String SQL_QUERY_SELECTALL = "SELECT id_template, name, content FROM workflow_task_formspdf_template";
+
+	 private static final String SQL_QUERY_SELECT = SQL_QUERY_SELECTALL + " WHERE id_template = ?";
+	 
+	 private static final String SQL_QUERY_INSERT = "INSERT INTO workflow_task_formspdf_template ( name, content ) VALUES ( ?, ? ) ";
+	 
+	 private static final String SQL_QUERY_DELETE = "DELETE FROM workflow_task_formspdf_template WHERE id_template = ? ";
+	 
+	 private static final String SQL_QUERY_UPDATE = "UPDATE workflow_task_formspdf_template SET name = ?, content = ? WHERE id_template = ?";
+	 
+	 @Override
+	 public void insert(FormsPDFTaskTemplate formsPDFTaskTemplate)
+	 {
+		try ( DAOUtil daoUtil = new DAOUtil( SQL_QUERY_INSERT, Statement.RETURN_GENERATED_KEYS, FormsPDFPlugin.getPlugin( ) ) )
+        {
+			int nIndex = 0;
+			daoUtil.setString( ++nIndex, formsPDFTaskTemplate.getName() );
+			daoUtil.setString( ++nIndex, formsPDFTaskTemplate.getContent() );
+			
+			daoUtil.executeUpdate( );
+			
+			if ( daoUtil.nextGeneratedKey( ) )
+			{
+				formsPDFTaskTemplate.setId( daoUtil.getGeneratedKeyInt( 1 ) );
+			}
+			daoUtil.free( );
+        }
+	 }
+	 
+	 @Override
+	 public void store(FormsPDFTaskTemplate formsPDFTaskTemplate)
```

Comment:
you must use try with resource in all dao method like you have done in insert method
delete all call of dao daoUtil.free( ); in clause try